File: 2083-three-divisors/three-divisors.java
class Solution {
    public boolean isThree(int n) {
        int r = (int) Math.sqrt(n);
        return r * r == n && isPrime(r);
    }

    private boolean isPrime(int n) {
        if (n < 2) return false;

        for (int i = 2; i * i <= n; i++) {
            if (n % i == 0) return false;
        }

        return true;
    }
}

File: 2083-three-divisors/three-divisors.py
class Solution:
    def isThree(self, n: int) -> bool:
        r = int(n ** 0.5)
        return r * r == n and self.isPrime(r)

    def isPrime(self, n: int) -> bool:
        if n < 2:
            return False
        for i in range(2, int(n ** 0.5) + 1):
            if n % i == 0:
                return False
        return True